src/persistence.js

```diff
@@ -332,48 +341,42 @@ export default function _Persistence(mpInstance) {
         return parsedDecodedProducts;
     };
 
-    this.setProductStorage = function() {
+    // https://go.mparticle.com/work/SQDSDKS-6021
+    this.setLocalStorage = function() {
         if (!mpInstance._Store.isLocalStorageAvailable) {
             return;
         }
 
-        var allLocalStorageProducts = self.getAllUserProductsFromLS(),
+        var key = mpInstance._Store.storageName,
+            allLocalStorageProducts = self.getAllUserProductsFromLS(),
+            localStorageData = self.getLocalStorage() || {},
             currentUser = mpInstance.Identity.getCurrentUser(),
             mpid = currentUser ? currentUser.getMPID() : null,
             currentUserProducts = {
                 cp: allLocalStorageProducts[mpid]
                     ? allLocalStorageProducts[mpid].cp
                     : [],
             };
+
+        if (mpInstance._Store.getPrivacyFlag('Products')) {
+            return;
+        }
         if (mpid) {
             allLocalStorageProducts = allLocalStorageProducts || {};
             allLocalStorageProducts[mpid] = currentUserProducts;
             try {
-                window.localStorage.setItem(
-                    encodeURIComponent(mpInstance._Store.prodStorageName),
-                    Base64.encode(JSON.stringify(allLocalStorageProducts))
-                );
+                if (!mpInstance._Store.getPrivacyFlag('Products')) {
+                    window.localStorage.setItem(
+                        encodeURIComponent(mpInstance._Store.prodStorageName),
+                        Base64.encode(JSON.stringify(allLocalStorageProducts))
+                    );
```

Comment:
I would also clean up the original code a bit so that i's more readable:

```suggestion
                    const encodedKey = encodeURIComponent(mpInstance._Store.prodStorageName);
                    const encodedValue = Base64.encode(JSON.stringify(allLocalStorageProducts));
                    window.localStorage.setItem(encodedKey, encodedValue);
```